openmc_plotter/main_window.py

```diff
@@ -1185,6 +1202,12 @@ def saveSettings(self):
         if len(self.model.subsequentViews) > 10:
             self.model.subsequentViews = self.model.subsequentViews[-10:]
 
+        # get hashes for cell/mat ids at close
+        self.model.cell_ids_hash = hashlib.md5(
+            pickle.dumps(self.model.cell_ids)).hexdigest()
+        self.model.mat_ids_hash = hashlib.md5(
+            pickle.dumps(self.model.mat_ids)).hexdigest()
```

Comment:
I don't think this really does what we want. @pshriwise can correct me if I'm wrong, but I believe `cell_ids` and `mat_ids` are the material/cell IDs for the currently visible plot. That doesn't really tell us whether or not the underlying model changed.

Reply:
Do you have an example of a change that would trigger the errors?

Reply:
Not offhand, but just reasoning about it -- if you had a case where a cell/material changed outside of the current view, the hash comparison based on this would not catch it. I think the best thing to do would be to get the hexdigest of the XML files themselves.

Reply:
Before doing this change, I had tried changing material compositions, surface values/locations, etc, in the XML files and it did not trigger any errors with loading the GUI. The only errors that were triggered were when there was new ID values. So basically, I think it already works in those situations.

Reply:
The `model` that gets loaded by the `plot_settings.pkl` file does technically also have `temperatures` and `densities` attributes, which you would think means that that info is also stored. However, those all seem to just be `nan` so none of that information is actually loaded. I think the only thing it stores and uses are these IDs and then gets the info from the XML files based on the ID. So so for example even if the material definition changes, but the IDs are still the same, there is no issue with loading and viewing.

Reply:
Hiya! I was just looking at some issues here and noticed this discussion. I like the use of the hexdigest as a quick way to compare the XML's. 

I think I understand @paulromano's concern here. The material and cell IDs that are saved to the `plot_settings.pkl` file in this PR represent the image data the user was viewing when the application was closed. If the model changes, there's no guarantee that the change would affect that view or that the newly created model is going to reproduce that same view. 

I think an XML-based hash with a life-cyle like the following is the best combination of ease and accuracy:

  - When the application opens, create a hash for the `geometry.xml` and `material.xml` files. Compare against the XML-based hash in the `plot_settings.pkl` file (if it exists)
  - While the application is running, update the XML hash if the model is re-loaded
  - Save the hash to the `plot_settins.pkl` when the application closes (should happen automatically as part of pickle-ing the model object

Reply:
I really think that checking the entire file is overkill. It will certainly prevent the error from happening, but I think there is still a lot that is allowed to be changed within the model that doesn't actually cause issue with the viewer when it reopens.

I haven't been able to reproduce any error personally, even with changing all sorts of views in the plotter to load new information, except for when the cell IDs and mat IDs are changed (either new IDs or change where they are located). But simply changing the dimensions of cells/surfaces and changing a material composition doesn't cause issues. If you add cells or new materials, then these lists are now different and the difference is caught. But I think we can still save a lot of reloading time for people if we minimize this to the minimum.

If anyone can give me an example where this error occurs, beyond what is currently checked, I can change this further.

Reply:
> But I think we can still save a lot of reloading time for people

Maybe, but at the expense of complexity in code. The nice thing about checking the hash of the XML files is that it is very simple and we don't have to think deeply about what changes are acceptable and what changes aren't. Regarding the time itself -- you might be surprised how fast you can hash a large file. I took a model of ITER (328k cells, 392k surfaces, 131k materials) and on my laptop getting a hash of both geometry.xml and materials.xml takes half a second. That is a small fraction of the time that it takes to actually generate a plot for such a large, complex model.

Reply:
Fair point that other values in the XMLs could change, but the plotter will incorporate those updates and work just fine. I'll will note that in the case we use an XML hash, the user can still reload the model from the GUI if they'd like to incorporate such changes.

> If you add cells or new materials, or change which cells contain which materials, then these lists are now different and the difference is caught before the error arises.

Looking at the code, the hashed set of IDs come from a particular plot view, so isn't it possible that IDs have changed in a region of the model which is not in the particular view used for this check?

I was trying to demonstrate this and I noticed that the material/cell IDs from the `plot_settings.pkl` is always comparing against the set of IDs generated by the default view, so I think that unless one closes the plotter on the default view, the settings file is going to be ignored.

Reply:
> Looking at the code, the hashed set of IDs come from a particular plot view, so isn't it possible that IDs have changed in a region of the model which is not in the particular view used for this check?

I will investigate how this is handled in a little more detail, but so far (I think) it seems to catch if things that are not a part of the view have changed.

> I was trying to demonstrate this and I noticed that the material/cell IDs from the plot_settings.pkl is always comparing against the set of IDs generated by the default view, so I think that unless one closes the plotter on the default view, the settings file is going to be ignored.

I have been using this current branch as I am looking at models and making different views. It has worked to reopen a model and have it in the last view before closing (ie, not the default starting view).